File: discounts/src/domain/Product.js
const camelCase = require('camelcase-keys');


class Product {
    constructor(params) {
        this.id = params.id
        this.externalId = params.externalId
        this.price = params.price
        this.title = params.title
        this.description = params.description
    }

    static fromDB(dbObject) {
        return new Product(camelCase(dbObject))
    }
}

module.exports = Product

File: products/src/main/kotlin/com/github/joaolucasl/chumlee/products/domain/Product.kt
package com.github.joaolucasl.chumlee.products.domain

data class Product(
    val id: Int,
    val externalId: String,
    var price: Int,
    val title: String,
    val description: String
) {
    fun applyDiscount(percentage: Float) {
        price -= (price * percentage).toInt()
    }
}